04.class/operateInterface.js

```diff
@@ -0,0 +1,88 @@
+import readline from "readline";
+import enquirer from "enquirer";
+
+const { Select } = enquirer;
+
+export default class OperateInterface {
+  constructor(memoOperation) {
+    this.memoOperation = memoOperation;
+  }
+
+  async listMemos() {
+    const memos = await this.memoOperation.fetchMemos();
+    if (this.#isMemoEmpty(memos)) return;
+    memos.forEach((memo) => {
+      console.log(memo.content.split("\n")[0]);
+    });
+  }
+
+  async readMemos() {
+    const content = await this.#selectMemoFromList("see");
+    if (!content) return;
+    console.log(content);
+  }
+
+  async deleteMemo() {
+    const id = await this.#selectMemoFromList("delete");
+    if (!id) return;
+    this.memoOperation.removeMemo(id);
+    console.log("Memo deleted!");
+  }
+
+  async writeMemo() {
+    const content = await this.#imputMemo();
+    await this.memoOperation.addMemo(content);
+    console.log("Memo added.");
+  }
+
+  async #selectMemoFromList(action) {
+    const memos = await this.memoOperation.fetchMemos();
+    if (this.#isMemoEmpty(memos)) return;
+    const choices = this.#extractFirstLineAndValue(memos, action);
+    return this.#promptForMemo(choices, action);
+  }
+
+  #isMemoEmpty(memos) {
+    if (memos.length === 0) {
+      console.error("No memos found.");
+      return true;
+    }
+  }
+
+  #extractFirstLineAndValue(memos, action) {
+    return memos.map((memo) => {
+      const firstLine = memo.content.split("\n")[0] || " ";
+      const prop = action === "delete" ? "id" : "content";
+      return { name: firstLine, value: memo[prop] };
+    });
+  }
+
+  #promptForMemo(memo, action) {
```

Comment:
`return this.#promptForMemo(choices, action);`と呼び出すならここは
```suggestion
  #promptForMemo(choices, action) {
```

にしましょう。（同じデータには同じ名前を付ける）

https://bootcamp.fjord.jp/pages/readable-variable-name#%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89%E3%81%AE%E5%BC%95%E6%95%B0%E3%82%82%E8%80%83%E3%81%88%E6%96%B9%E3%81%AF%E5%90%8C%E3%81%98

Reply:
こちらも確認不足でした💦
ご指摘ありがとうございます。
修正いたします！